### PR TITLE
chore(deps): eject create-react-app and upgrade Jest

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - we want a version of Jest that can handle package.json's "exports"

This commit:
 - eject create-react-app from fxa-settings
 - upgrade Jest to the latest version in fxa-settings

### CODE REVIEW

**Comment by chenba** [comment]
Note that I plan to eject/upgrade each package that uses CRA separately instead of one big PR.

**Comment by chenba** [comment]
@dschom, @LZoog, @vpomerleau, and/or @vbudhram r?  Most of the changes are from copied files when ejecting CRA.

File: packages/fxa-settings/config/env.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+'use strict';
```
Comment (vbudhram) [inline packages/fxa-settings/config/env.js:1]: Will need to add license header to the files.

```
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment (chenba) [reply packages/fxa-settings/config/env.js:1]: hmm this is one of the many files that the eject script copied over, so I have no idea whether we should or need to put the Moz license header on them.  They are actually from react-scripts after all.

**Comment by chenba** [comment]
Latest push was only to resolve conflicts.

File: packages/fxa-settings/src/react-app-env.d.ts [hunk 2]
```diff
@@ -1,9 +1,79 @@
...
+  const classes: { readonly [key: string]: string };
+  export default classes;
+}
+
+declare module '*.module.sass' {
+  const classes: { readonly [key: string]: string };
+  export default classes;
+}
+
 /* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (vpomerleau) [inline packages/fxa-settings/src/react-app-env.d.ts:73]: Should L73-79 be moved to the top of the file? I assume the previous line were added automatically when ejecting.

File: packages/fxa-settings/config/env.js (hunk 1 above)
Comment (vpomerleau) [reply packages/fxa-settings/config/env.js:1]: Should we instead add a header to indicate that these are from react-scripts?

**Review by vpomerleau — COMMENTED** [review]
As far as I can tell from checking this out locally (running tests, starting storybook, going through a few flows in /settings), the app looks to be functioning as expected with these changes. (except for that one stray `import '@testing-library/jest-dom/extend-expect'` in BrandingMessage's test file - I assume that got pulled in with a rebase).

It was a bit hard to tell which changes where done manually and which were automatically copied. If there's anything in particular that would be good to review, please let me know.

File: packages/fxa-settings/config/env.js (hunk 1 above)
Comment (chenba) [reply packages/fxa-settings/config/env.js:1]: > Should we instead add a header to indicate that these are from react-scripts?

Sounds good.  I'll add a comment but not a license header.

**Comment by chenba** [comment]
> (except for that one stray `import '@testing-library/jest-dom/extend-expect'` in BrandingMessage's test file - I assume that got pulled in with a rebase).

heh yeah exactly.  It's been fixed.

> It was a bit hard to tell which changes where done manually and which were automatically copied. If there's anything in particular that would be good to review, please let me know.

Sorry about that; I think I just kinda ran out of time or got distracted on Friday.  I'll add comments pointing out where some of the manual edits in the copied files are.

File: packages/fxa-settings/config/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+  try {
+    require.resolve('react/jsx-runtime');
+    return true;
+  } catch (e) {
+    return false;
+  }
+})();
+
+const allFxa = path.resolve(__dirname, '../../');
+const importPaths = [allFxa, path.resolve(__dirname, '../../../node_modules')];
```
Comment (chenba) [inline packages/fxa-settings/config/webpack.config.js:91]: This is inlining what we used to do with `rescripts`; used on lines 336 and 417 below.

**Review by vbudhram — APPROVED** [review]
@chenba Thanks for adding headers and flagging generated files. Feel pretty good that the test pass, lets try it!

**Pull request merged**